test/IntegrationTests/TicketService/test_list_custom_fields.py

```diff
@@ -0,0 +1,190 @@
+            }
+            response = requests.post(f"{self.BASE_URL}/custom_fields", json=data)
+            if response.status_code != HTTPStatus.CREATED:
+                logger.error(f"Failed to add field: {field['field_name']}. Server response: {response.text}")
+            self.assertEqual(response.status_code, HTTPStatus.CREATED,
+                             f"Failed to add field: {field['field_name']}")
+
+        # Fetch the custom fields for the customer again after adding
+        url = f"{self.BASE_URL}/custom_fields?customer_guid={self.valid_customer_guid}"
+        response = requests.get(url)
+        self.assertEqual(response.status_code, HTTPStatus.OK, "Failed to list custom fields")
+        response_data = response.json()
+
+        # Log the actual response data to see what fields are returned
+        logger.info(f"Response data: {response_data}")
+
+        # Verify all added fields are present
+        self.assertEqual(len(response_data), len(custom_fields), "Mismatch in number of custom fields")
+        for expected_field in custom_fields:
+            matching_field = next(
+                (field for field in response_data if field["field_name"] == expected_field["field_name"]), None)
+            self.assertIsNotNone(matching_field, f"Custom field not found: {expected_field['field_name']}")
+            self.assertEqual(matching_field["field_type"], expected_field["field_type"],
+                             f"Field type mismatch for {expected_field['field_name']}")
+            self.assertEqual(matching_field["required"], expected_field["required"],
+                             f"Required flag mismatch for {expected_field['field_name']}")
+
+        logger.info("Test for multiple custom fields of all types passed.")
+
+    def _add_50_custom_fields(self):
+        """Test adding 50 custom fields to a customer."""
+        logger.info("Adding 50 custom fields to the customer.")
+
+        # Create a list of 50 custom fields with alternating required values
+        custom_fields = [{"field_name": f"field_{i}", "field_type": "VARCHAR(255)", "required": i % 2 == 0} for i in
+                         range(50)]
+
+        # Iterate over each field and send a POST request to add it
+        for field in custom_fields:
+            data = {
+                "customer_guid": self.valid_customer_guid,
+                **field
+            }
+            response = requests.post(f"{self.BASE_URL}/custom_fields", json=data)
+
+            # Log and assert successful creation of each field
+            if response.status_code != HTTPStatus.CREATED:
+                logger.error(f"Failed to add field: {field['field_name']}. Server response: {response.text}")
+            self.assertEqual(response.status_code, HTTPStatus.CREATED,
+                             f"Failed to add field: {field['field_name']}")
+
+        logger.info("Successfully added 50 custom fields.")
+
+    def test_pagination_for_custom_fields(self):
```

Comment:
Check for getting data for this testcase with page_number = 1 and page_size = 60 and verify only 50 records are replied back

Dont resolve this comment
Just reply to this comment that you have fixed it

Reply:
Resolved sir